superset/assets/javascripts/dashboard/components/DashboardViewContainer.jsx

```diff
@@ -0,0 +1,305 @@
+  }
+
+  updateDashboardLayout(layout) {
+    this.props.actions.updateDashboardLayout(layout);
+  }
+
+  addSlicesToDashboard(sliceIds) {
+    return this.props.actions.addSlicesToDashboard(this.props.dashboard.id, sliceIds);
+  }
+
+  toggleExpandSlice(slice, isExpanded) {
+    this.props.actions.toggleExpandSlice(slice, !isExpanded);
+  }
+
+  addFilter(sliceId, col, vals, merge, refresh) {
+    this.props.actions.addFilter(sliceId, col, vals, merge, refresh);
+  }
+
+  clearFilter(sliceId) {
+    this.props.actions.clearFilter(sliceId);
+  }
+
+  removeFilter(sliceId, col, vals) {
+    this.props.actions.removeFilter(sliceId, col, vals);
+  }
+
+  bindResizeToWindowResize() {
+    let resizeTimer;
+    const dash = this;
+    const allSlices = this.getAllSlices();
+    $(window).on('resize', () => {
+      clearTimeout(resizeTimer);
+      resizeTimer = setTimeout(dash.renderSlices(allSlices), 500);
+    });
+  }
+
+  // render an list of slices
+  renderSlices(slc, force = false, interval = 0) {
+    const dash = this;
+    const slices = slc || this.getAllSlices();
+    if (!interval) {
+      slices.forEach(slice => (dash.fetchSlice(slice, force)));
+      return;
+    }
+
+    const meta = this.props.dashboard.metadata;
+    const refreshTime = Math.max(interval, meta.stagger_time || 5000); // default 5 seconds
+    if (typeof meta.stagger_refresh !== 'boolean') {
+      meta.stagger_refresh = meta.stagger_refresh === undefined ?
+        true : meta.stagger_refresh === 'true';
+    }
+    const delay = meta.stagger_refresh ? refreshTime / (slices.length - 1) : 0;
+    slices.forEach((slice, i) => {
+      setTimeout(() => dash.fetchSlice(slice, force), delay * i);
+    });
+  }
+
+  render() {
+    return (
+      <Dashboard
+        initMessages={this.props.initMessages}
+        dashboard={this.props.dashboard}
+        slices={this.props.slices}
+        datasources={this.props.datasources}
+        filters={this.props.filters}
+        refresh={this.props.refresh}
+        timeout={this.props.timeout}
+        user_id={this.props.user_id}
+        isStarred={this.props.isStarred}
+        getFormDataExtra={this.getFormDataExtra.bind(this)}
+        fetchSlice={this.fetchSlice.bind(this)}
+        renderSlices={this.renderSlices.bind(this, this.getAllSlices())}
+        startPeriodicRender={this.startPeriodicRender.bind(this)}
+        updateDashboardTitle={this.updateDashboardTitle.bind(this)}
+        readFilters={this.readFilters.bind(this)}
+        fetchFaveStar={this.fetchFaveStar.bind(this)}
+        saveFaveStar={this.saveFaveStar.bind(this)}
+        saveSlice={this.saveSlice.bind(this)}
+        removeSlice={this.removeSlice.bind(this)}
+        removeChart={this.removeChart.bind(this)}
+        updateDashboardLayout={this.updateDashboardLayout.bind(this)}
+        addSlicesToDashboard={this.addSlicesToDashboard.bind(this)}
+        toggleExpandSlice={this.toggleExpandSlice.bind(this)}
```

Comment:
couldn't you just:
`toggleExpandSlice={this.props.actions.toggleExpandSlice}` here?

Also to avoid much boilerplate here I wouldn't be against having something like `dashboardActions` and pass that to `<Dashboard>`. Could also go in the direction of reusable and composable actions/reducers

Reply:
Fixed. I prefer to use `Callback Functions as Props` instead of passing action creators itself down to children components. It will be easier to test child component by just mocking callbacks. Also It's one of good practice mentioned in Airbnb redux/best-practices, as well as outside airbnb:
http://www.thegreatcodeadventure.com/the-react-plus-redux-container-pattern/